geoh5py/io/h5_reader.py

```diff
@@ -181,21 +181,23 @@ def fetch_concatenated_values(
                 group = h5file[name][entity_type][as_str_if_uuid(uid)][
                     "Concatenated Data"
                 ]
-                if label not in group["Index"]:
+                indices = group["Index"].get(label.replace("/", "\u2044"))
+                if indices is None:
                     return None
 
```

Comment:
strange to change "/" by unicode but trusting you on this one

Reply:
Has to do with h5py using the "/" as a database seperator, which breaks the structure of the file if we don't replace.